polkadot/xcm/xcm-executor/src/traits/asset_exchange.rs

```diff
@@ -37,6 +37,13 @@ pub trait AssetExchange {
 		want: &Assets,
 		maximal: bool,
 	) -> Result<AssetsInHolding, AssetsInHolding>;
+
+	/// Handler for quoting the exchange price of two assets.
+	///
+	/// - `asset1` The first asset.
+	/// - `asset2` The second asset.
+	/// - `maximal`: If `true`, then all of `asset1` should be used.
+	fn quote_exchange_price(asset1: &Asset, asset2: &Asset, maximal: bool) -> Option<u128>;
```

Comment:
I don't think it makes sense to accept collections of assets here.

The API was meant to mimic the `QuotePrice` trait, but using `maximal: bool` to distinguish the two cases.

Reply:
I also think the non-symmetricity with `exchange_asset` makes it difficult to use in more general cases.

Let's say you want to quote the price for 2 assets, from 2 other assets, to later call `exchange_asset` (a more general case than the current PR). Then there is no way to use this `quote_exchange_price` to get the expected requirement for `give` to get a successful execution of `exchange_asset`.

maybe something like this could be better
 ```rust
/// Returns a vector of minimal assets in holding which would execute
/// a successful `exchange_asset(_, _, give, want, _)`
fn exchange_asset_minimal_needed(give: &[AssetId], want: &Assets) -> Vec<AssetsInHolding>
```
Note that we could also support non-fungibles with a more complex `give` parameter maybe.
I'm not sure if we should also give the `Option<Location>` information

Then the tuple implementation would just be:
```rust
{
	for_tuples!( #(
		Tuple::exchange_asset_minimal_needed(give, want)
	)* )
		.flatten()
}
```